analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -320,14 +320,19 @@ public CSharpRemovableSymbolWalker(Func<SyntaxTree, bool, SemanticModel> getSema
                 this.containingTypeAccessibility = containingTypeAccessibility;
             }
 
-            // This override is needed because VisitRecordDeclaration is not available due to the Roslyn version.
+            // This override is needed because VisitRecordDeclaration and LocalFunctionStatementSyntax are not available due to the Roslyn version.
             public override void Visit(SyntaxNode node)
             {
                 if (node.IsAnyKind(SyntaxKindEx.RecordClassDeclaration, SyntaxKindEx.RecordStructDeclaration))
                 {
                     VisitBaseTypeDeclaration(node);
                 }
 
+                if (LocalFunctionStatementSyntaxWrapper.IsInstance(node))
```

Comment:
```suggestion
                if (node.IsKind(SyntaxKindEx.LocalFunctionStatement))
```
This seems more straightforward (and consistent with the rest of the implementation.